lib/plugin-installer.js

```diff
@@ -0,0 +1,272 @@
+const fs = require('fs');
+const os = require('os');
+const path = require('path');
+const { spawn } = require('child_process');
+const YAWN = require('yawn-yaml/cjs');
+
+class PluginInstaller {
+  constructor (options) {
+    options = options || {};
+    this.packageName = options.packageName || null;
+    this.pluginManifest = options.pluginManifest || null;
+    this.config = options.config;
+  }
+
+  static get PACKAGE_PREFIX () {
+    return 'express-gateway-plugin-';
+  }
+
+  static create (options) {
+    return new PluginInstaller(options);
+  }
+
+  runNPMInstallation ({ packageSpecifier, cwd, env }) {
+    return new Promise((resolve, reject) => {
```

Comment:
Sure, take a look.  Let me know if you find anything.  I don't see why this would be frowned upon.  The only events we'd need to communicate are based on success or failure, which surface through the Promise API quite nicely.